PatternPal/PatternPal.Extension/PatternPal.ExtensionPackage.cs

```diff
@@ -135,12 +136,29 @@ public bool DoLogData
             get { return _doLogData; }
             set
             {
+                // Set is triggered both when changing the field value, as well as when clicking on the OK button.
+                // This prevents the code from being triggered twice.
+                if (value == _doLogData) return;    
+
                 ThreadHelper.JoinableTaskFactory.Run(async () => {
                     await (value
                         ? SubscribeEvents.SubscribeEventHandlersAsync()
                             : SubscribeEvents.UnsubscribeEventHandlersAsync());
                 });
+
+                // The SessionId has to be reset if the option for logging data is changed to prevent logging without a session id. 
+                // In other words, a new session has to be started.
+                if (!_doLogData && value)
```

Comment:
On line 141 you check that `value == _doLogData`, so you don't have to check both values here, as they'll never be the same. Instead, you can just look at `value` to decide whether a session is started or ended.